production build and run tasks, to simplify production HOWTO docs

boutell opened the pull request.

boutell commented:
These are compatible with the steps for a3-assembly-boilerplate. I would like to kill `npm run dev` like I did there too and get to a simpler nodemon startup etc., but I don't need that for this HOWTO.

abea commented on package.json:
```diff
@@ -5,8 +5,9 @@
-    "dev": "nodemon --delay 1000ms -x \"nodemon app.js\""
+    "dev": "nodemon --delay 1000ms -x \"nodemon app.js\"",
+    "build": "NODE_ENV=production node app @apostrophecms/asset:build",
+    "production-start": "NODE_ENV=production node app"
```
I think `npm run prod` is [more common](https://github.com/laravel/laravel/blob/8.x/package.json) and feels cleaner. If we're to start having a standard command for production I suggest that be it.

boutell commented:
So far the only mention I've found of "npm run prod" is for Laravel, and I
think they use it for something that's more like "build" than "run."

On Tue, Jul 6, 2021 at 9:49 AM Alex Bea ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
> ------------------------------
>
> In package.json
> <https://github.com/apostrophecms/a3-boilerplate/pull/25#discussion_r664572772>
> :
>
> > @@ -5,8 +5,9 @@
>    "main": "app.js",
>    "scripts": {
>      "start": "node app",
> -    "build": "node app @apostrophecms/asset:build",
> -    "dev": "nodemon --delay 1000ms -x \"nodemon app.js\""
> +    "dev": "nodemon --delay 1000ms -x \"nodemon app.js\"",
> +    "build": "NODE_ENV=production node app @apostrophecms/asset:build",
> +    "production-start": "NODE_ENV=production node app"
>
> I think npm run prod is more common
> <https://github.com/laravel/laravel/blob/8.x/package.json> and feels
> cleaner. If we're to start having a standard command for production I
> suggest that be it.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apostrophecms/a3-boilerplate/pull/25#pullrequestreview-699977011>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAH27JCRMLB67FV4MKHH2TTWMCVXANCNFSM47XN27ZA>
> .
>


-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his

abea commented:
Let's use `npm start` for this, then. It's not doing anything interesting at the moment. Not much reason to use it since in each environment there's a more specific thing. `npm run production-start` seems like an unnecessary mouthful.

abea commented:
Others use `npm run serve`, though I'm not sure if that applies as much.

boutell commented:
I didn't use use `npm start` because `nodemon` really, really wants to, and nodemon is for dev. I see that I hadn't brought it over yet, but nodemon configuration is greatly simplified if we let `npm start` be suitable for dev. Now I've updated the PR with `dev` just being `nodemon`.

`npm run serve` feels good to me, that's in this update too.

Finally I added `npm run release` which takes care of all the things you have to do after you `git pull` updated code for your site. I did not use `update` because that would be very confusing due to `npm update` being a very different beast. `release` is a good name because it constitutes a new release of your site.
